assets/prototype/domain/eventEditor/datetimes/dateList/filterBar/useDatesListFilterState.test.ts

```diff
@@ -0,0 +1,324 @@
+	});
+	expect(result.current.displayDates).toBe('both');
+});
+
+test('should update showDates by invoking setShowDates with corresponding accepted enums', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setShowDates(ShowDates.all);
+	});
+	expect(result.current.showDates).toBe('all');
+
+	act(() => {
+		result.current.setShowDates(ShowDates.upcomingOnly);
+	});
+	expect(result.current.showDates).toBe('upcomingOnly');
+});
+
+test('should update processedDates to reflect changes made by invoking setDatesSortedBy', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.chronologically);
+	});
+	const startDates = R.map(R.view(R.lensProp('startDate')))(result.current.processedDates);
+	const expectedstartDates = [
+		'2000-01-12T08:00:00+00:00',
+		'2009-12-18T11:31:00+00:00',
+		'2019-12-18T11:31:00+00:00',
+		'2020-01-12T08:00:00+00:00',
+	];
+	expect(startDates).toEqual(expectedstartDates);
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byId);
+	});
+	const ids = R.map(R.view(R.lensProp('id')))(result.current.processedDates);
+	const expectedIds = ['RGF0ZXRpbWU6ODM=', 'RGF0ZXRpbWU6ODU=', 'RGF0ZXRpbWU6ODU=', 'WGF0ZXRpbWU6ODM='];
+	expect(ids).toEqual(expectedIds);
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byName);
+	});
+	const names = R.map(R.view(R.lensProp('name')))(result.current.processedDates);
+	const expectedNames = ['another title', 'just another datetime', 'test', 'test'];
+	expect(names).toEqual(expectedNames);
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byOrder);
+	});
+	const datesOrder = R.map(R.view(R.lensProp('order')))(result.current.processedDates);
+	const expectedDatesOrder = [-2, -1, 1, 1];
+	expect(datesOrder).toEqual(expectedDatesOrder);
+});
+
+test('should update processedDates to reflect changes made by invoking setShowDates with activeOnly filter', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setShowDates(ShowDates.activeOnly);
+	});
+	const dates = result.current.processedDates;
+
+	expect(dates.length).toBe(2);
+	expect(dates[0].isActive).toBe(true);
+	expect(dates[1].isActive).toBe(true);
+});
+
+test('should update processedDates to reflect changes made by invoking setShowDates with activeUpcoming filter', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setShowDates(ShowDates.activeUpcoming);
+	});
+	const dates = result.current.processedDates;
+
+	expect(dates.length).toBe(3);
+	expect(dates[0].isActive).toBe(true);
+	expect(dates[0].isUpcoming).toBe(true);
+	expect(dates[1].isActive).toBe(true);
+	expect(dates[1].isUpcoming).toBe(false);
+	expect(dates[2].isActive).toBe(false);
+	expect(dates[2].isUpcoming).toBe(true);
```

Comment:
This is also error prone if the `datetimes` array changes. It could be like this 
```js
expect(dates[0].isActive).toBe(datetimes[0].isActive);
```

Reply:
Good suggestion, thx.

Reply:
As shown in slack, the order isn't as original, I can't do that @manzoorwanijk :cry: 
![image](https://user-images.githubusercontent.com/1477453/71525972-041b4180-28dd-11ea-8a85-e84909d4890d.png)